kie-dmn-feel/src/main/java/org/kie/dmn/feel/runtime/functions/BaseFEELFunction.java

```diff
@@ -273,7 +274,7 @@ private void adjustForVariableParameters(CandidateMethod cm, Class<?>[] paramete
 
     private Object normalizeResult(Object result) {
         // this is to normalize types returned by external functions
-        return result != null && result instanceof Number && !(result instanceof BigDecimal) ? new BigDecimal( result.toString() ) : result;
```

Comment:
I've noticed one last stray creation of BigDecimal in the `functions` package here, and I've normalized this to the `EvalHelper` too.